geo-types/src/algorithms.rs

```diff
@@ -0,0 +1,94 @@
+// FIXME: everything in this file is copy/paste from the 'geo' crate. ideally we
+//        wouldn't have this duplication
```

Comment:
the (non-public) traits below are unfortunately necessary to keep the `spade` trait `impl`s on the geo types in `geo-types/src/lib.rs`. for now it's a little messy, but i think we can reevaluate this in the future. i.e. do we really need `spade`?